api-server/server/boot/donate.js

```diff
@@ -246,24 +258,89 @@ export default function donateBoot(app, done) {
 
   const pubKey = keys.stripe.public;
   const secKey = keys.stripe.secret;
+  const paypalKey = keys.paypal.client;
+  const paypalSec = keys.paypal.secret;
   const hmacKey = keys.servicebot.hmacKey;
   const secretInvalid = !secKey || secKey === 'sk_from_stripe_dashboard';
   const publicInvalid = !pubKey || pubKey === 'pk_from_stripe_dashboard';
+
+  const paypalSecretInvalid =
+    !paypalKey || paypalKey === 'id_from_paypal_dashboard';
+  const paypalPublicInvalid =
+    !paypalSec || paypalSec === 'secret_from_paypal_dashboard';
   const hmacKeyInvalid =
     !hmacKey || hmacKey === 'secret_key_from_servicebot_dashboard';
+  const paypalInvalid = paypalPublicInvalid || paypalSecretInvalid;
+  const stripeInvalid = secretInvalid || publicInvalid || hmacKeyInvalid;
 
-  if (secretInvalid || publicInvalid || hmacKeyInvalid) {
+  if (stripeInvalid) {
     if (process.env.FREECODECAMP_NODE_ENV === 'production') {
       throw new Error('Stripe API keys are required to boot the server!');
     }
     console.info('No Stripe API keys were found, moving on...');
-    done();
```

Comment:
`done` is being removed, where is this handled?

Reply:
It's handled down here:
https://github.com/freeCodeCamp/freeCodeCamp/blob/d62e37b2aaaa421ea1855d0a02bfdc8a3bc7418a/api-server/server/boot/donate.js#L341-L344

I think we should move `addDonation` and `updatePaypal` out of the way, since defining them in the middle of `donateBoot` makes it harder to follow the logic.